packages/btcindexer/src/btcindexer.ts

```diff
@@ -54,4 +58,80 @@ export class Indexer {
 
 		return true;
 	}
+
+	async scanNewBlocks(): Promise<void> {
+		const blocksToProcess = await this.d1
+			.prepare("SELECT height, hash FROM processed_blocks ORDER BY height ASC LIMIT 10")
+			.all<{ height: number; hash: string }>();
+
+		if (!blocksToProcess.results || blocksToProcess.results.length === 0) {
+			return;
+		}
+
+		const nbtcTxStatements: D1PreparedStatement[] = [];
+
+		const insertNbtcTxStmt = this.d1.prepare(
+			"INSERT INTO nbtc_txs (tx_id, hash, height, vout, sui_recipient, amount_sats, status) VALUES (?, ?, ?, ?, ?, ?, ?)",
+		);
+
+		for (const blockInfo of blocksToProcess.results) {
+			const rawBlockBuffer = await this.blocksDB.get(blockInfo.hash, {
+				type: "arrayBuffer",
+			});
+			if (!rawBlockBuffer) {
+				continue;
+			}
+			const block = Block.fromBuffer(Buffer.from(rawBlockBuffer));
+
+			for (const tx of block.transactions ?? []) {
+				const deposits = this.findNbtcDeposits(tx);
+				for (const deposit of deposits) {
+					nbtcTxStatements.push(
+						insertNbtcTxStmt.bind(
+							tx.getId(),
+							blockInfo.hash,
+							blockInfo.height,
+							deposit.vout,
+							deposit.suiRecipient,
+							deposit.amountSats,
+							"confirming",
+						),
+					);
+				}
+			}
+		}
+
+		// TODO: what happens if the array is empty?
+		await this.d1.batch(nbtcTxStatements);
```

Comment:
how about:
```
if (nbtcTxStatements.length > 0)
    await this.d1.batch(nbtcTxStatements);
```
or you can consider an early return if we don't need to delete anything.